client/src/components/header/header.component.tsx

```diff
@@ -42,9 +37,8 @@ class Header extends React.Component<SettingsActions> {
     );
   }
   
-  private showSidebar (): void {
-    this.props.setSidebarDisplayStatus(true);
-  }
+  private showSidebar = (): void => 
```

Comment:
What for you have replaced bindings to arrow functions? I know that code you are saving some space on your hard drive, but it become more unreadable if you do so.

Reply:
@appbak3r Well as far as I know arrow functions for methods which interacts state is standard code style for react at first place. In second place if you want add some extra methods you just add them and not bind. On third place commonly you can exlude "constructor" from class notation.

Reply:
@builat 

There is no standard code style for react as I know
There 2 points for not using arrow functions instead of bindings 
1. In arrow functions there is no context, and it is bad if you are manipulating with method, property modifiers, e.g: static. Context is unpredictable.
2. Another point that code becomes untestable, as I think, because I don't know the way how to spy on properties. If you know - welcome will be glad to know

Reply:
@appbak3r 

1. Arrow function takes context of creation block. And have no own context. So arrow function is equivalent for bind in this case.
2. About what properties you talking about?

Reply:
Just imagine some code that can be like this but in real
```js
class Foo {
   static bar = () => this;

   static foobar () {
    	   return this;
        }
}

function Bar () {
    class FooBar {
    	static bar = () => this;
    }
  
   console.log(FooBar.bar());
}

console.log(Foo.bar())
console.log(Foo.foobar())

new Bar();
```
What are will be results of output? As for me it is pretty unpredictable and unreadable. I know that no one will write such a code, but what if?

2. In Enzyme for testing private function you can just check if method was called on e,g button click, well you should create a spy on method to proxy it as a spy object of enzyme to count calls of that method.

Reply:
@appbak3r 
1. About your example. I think it will not pass code revew at first place. And I realy can't imagine where could be code like this used. Parts of code where I use arrow notation is quite simple and obvious.

2. In this case you probably right. Just read enzyme spec for this case. Here the hackaround https://medium.com/kevin-salters-blog/testing-react-with-enzyme-fbfc30190e70

Reply:
There is an article giving more points to not to use arrow functions instead of bound functions https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1

It completely describes my thoughts